src/Microsoft.DotNet.PlatformAbstractions/Native/PlatformApis.cs

```diff
@@ -112,11 +112,29 @@ private static DistroInfo LoadDistroInfo()
                         result.VersionId = line.Substring(11).Trim('"', '\'');
                     }
                 }
-                return result;
+
+                return NormalizeDistroInfo(result);
             }
             return null;
         }
 
+        // For some distros, we don't want to use the full version from VERSION_ID. One example is
+        // Red Hat Enterprise Linux, which includes a minor version in their VERSION_ID but minor
+        // versions are backwards compatable.
+        //
+        // In this case, we'll normalized RIDs like 'rhel.7.2' and 'rhel.7.3' to a generic
+        // 'rhel.7'. This brings RHEL in line with other distros like CentOS, Ubuntu or Debian which
+        // don't put minor version numbers in their VERSION_ID fields.
+        private static DistroInfo NormalizeDistroInfo(DistroInfo distroInfo)
+        {
+            if (distroInfo.Id == "rhel" && distroInfo.VersionId.IndexOf('.') != -1)
```

Comment:
(minor) - cache `distroInfo.VersionId.IndexOf('.')` instead of calling it twice.